src/frontend/extension.ts

```diff
@@ -1,28 +1,28 @@
-import * as vscode from "vscode";
-import * as net from "net";
-import * as fs from "fs";
-import * as path from "path";
-import * as os from "os";
+import { ExtensionContext, workspace, commands, window, Uri, TextDocumentContentProvider, CancellationToken } from "vscode";
+import { connect } from "net";
+import { existsSync, readdir } from "fs";
+import { extname, basename, join } from "path";
+import { tmpdir } from "os";
 
-export function activate(context: vscode.ExtensionContext) {
-	context.subscriptions.push(vscode.workspace.registerTextDocumentContentProvider("debugmemory", new MemoryContentProvider()));
-	context.subscriptions.push(vscode.commands.registerCommand("code-debug.examineMemoryLocation", examineMemory));
-	context.subscriptions.push(vscode.commands.registerCommand("code-debug.getFileNameNoExt", () => {
-		if (!vscode.window.activeTextEditor || !vscode.window.activeTextEditor.document || !vscode.window.activeTextEditor.document.fileName) {
-			vscode.window.showErrorMessage("No editor with valid file name active");
+export function activate(context: ExtensionContext) {
```

Comment:
keep the `vscode` import, it's the recommended style from the vscode docs and also makes finding symbols much easier than global imports. In fact I think all of these imports that you changed are much better in a namespace and shouldn't be unwrapped like this. Especially generic stuff like `join` or `connect` doesn't make much sense to import on its own and just makes the caller code less readable.